test/emoji.js

```diff
@@ -110,4 +110,24 @@ describe("emoji.js", function () {
       matchingEmojis.length.should.be.exactly(0);
     });
   });
+
+  describe("unemojify(str)", function () {
+    it("should parse emoji and replace them with :emoji:", function() {
+      var coffee = emoji.unemojify('I ❤️  ☕️! -  😯⭐️😍  ::: test : : 👍+');
+      should.exist(coffee);
+      coffee.should.be.exactly('I :heart:  :coffee:! -  :hushed::star::heart_eyes:  ::: test : : :+1:+');
+    })
+
+    it("should leave unknown emoji", function () {
+      var coffee = emoji.unemojify('I ⭐️ :another_one: 🥕');
+      should.exist(coffee);
+      coffee.should.be.exactly('I :star: :another_one: 🥕');
```

Comment:
Right now it only unemojifies the ones present in emoji.json. If carrot is added to that, this will be resolved automatically.